beefy-gadget/src/worker.rs

```diff
@@ -402,3 +384,133 @@ where
 
 	header.digest().convert_first(|l| l.try_to(id).and_then(filter))
 }
+
+/// Calculate the candidate block for the next BEEY vote.
+///
+/// Note, `number` is only used for better tracing output.
+fn vote_candidate<B>(
+	number: NumberFor<B>,
+	best_grandpa: NumberFor<B>,
+	best_beefy: NumberFor<B>,
+	min_delta: u32,
+) -> NumberFor<B>
+where
+	B: Block,
+{
+	let diff = best_grandpa.saturating_sub(best_beefy);
+	let diff = diff.saturated_into::<u32>();
+	let candidate = best_beefy + min_delta.max(diff.next_power_of_two()).into();
+
+	trace!(
+		target: "beefy",
+		"🥩 should_vote_on: #{:?}, diff: {:?}, next_power_of_two: {:?}, next_block_to_vote_on: #{:?}",
+		number,
+		diff,
+		diff.next_power_of_two(),
+		candidate,
+	);
+
+	candidate
+}
+
+#[cfg(test)]
+mod tests {
+	use super::vote_candidate;
+	use sp_runtime::testing::{Block, ExtrinsicWrapper, Header};
+
+	type MockBlock = Block<ExtrinsicWrapper<u64>>;
+
+	macro_rules! block {
+		($n:expr) => {
+			Header::new_from_number($n).number;
+		};
+	}
```

Comment:
I'm confused a bit, why is it a macro and not regular function? Also why do we create header and then only take `number` from it?
Wouldn't something like this work:
```rust
fn block(n: u32) -> NumberFor<MockBlock> {
  n.into()
}
```

Reply:
Actually finding this test header thing took me ages so did the first thing that came to mind. And I like macros ;-)

Reply:
Hmm, but this code can really be minimized to `$n.into()`, can't it? I'm trying to avoid macros cause they are magical (like now we actually have to discuss what the code is supposed to do), but can live with it as syntactic sugar, however if it can be minimized, let's do it.

Reply:
Agreed, the macro really isn't adding any value here.